src/dscanner/analysis/always_curly.d

```diff
@@ -0,0 +1,168 @@
+// Distributed under the Boost Software License, Version 1.0.
+//	  (See accompanying file LICENSE_1_0.txt or copy at
+//			http://www.boost.org/LICENSE_1_0.txt)
+
+module dscanner.analysis.always_curly;
+
+import dparse.lexer;
+import dparse.ast;
+import dscanner.analysis.base;
+import dsymbol.scope_ : Scope;
+
+import std.array : back, front;
+import std.algorithm;
+import std.range;
+import std.stdio;
+
+final class AlwaysCurlyCheck : BaseAnalyzer
+{
+	mixin AnalyzerInfo!"always_curly_check";
+
+	alias visit = BaseAnalyzer.visit;
+
+	///
+	this(string fileName, const(Token)[] tokens, bool skipTests = false)
+	{
+		super(fileName, null, skipTests);
+	}
+
+	void test(L,B)(L loc, B s)
+	{
+		if(!is(s == BlockStatement))
+		{
+			if(!s.tokens.empty) {
+				AutoFix af;
+				af.insertionBefore(s.tokens.front, "{");
+				af.insertionAfter(s.tokens.back, "{");
```

Comment:
```suggestion
				af.insertionAfter(s.tokens.back, "}");
```

probably? Also we don't have any way to specify the formatting style yet, so I'd recommend doing it all in allman style (there is one autofix thing in analysis that actually already generates indentation I think, you can look at that)

Reply:
:+1: